client/my-sites/checkout/checkout-thank-you/header.jsx

```diff
@@ -511,21 +508,45 @@ export class CheckoutThankYouHeader extends PureComponent {
 	}
 
 	getHeaderText() {
-		return getHeading( {
-			...this.props,
-			isSearch: this.isSearch(),
-		} );
+		const { isDataLoaded, hasFailedPurchases, primaryPurchase, translate } = this.props;
+
+		if ( ! isDataLoaded ) {
```

Comment:
Not critical, but I would use `if ( isDataLoading ) {`

Since now it looks a bit confusing, the condition below is literally - `isDataLoaded && hasFailedPurchases`, but actually it's not loaded, it's failed . Hope you get what I mean :) 
But not critical, up to you.

UPDATE: Oh, I see the method... ignore this my comment :)

Reply:
Also noting that, ultimately, we're looking to clean up (or remove) `header.jsx` at some point. Currently, it's only being used for legacy congrats pages that aren't redesigned.